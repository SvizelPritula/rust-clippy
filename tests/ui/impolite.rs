#![warn(clippy::impolite)]

trait Greeter {
    fn greet(&self);
}

struct HelloWorld;

impl Greeter for HelloWorld {
    fn greet(&self) {
        println!("Hello, world!");
    }
}

fn greet() {
    HelloWorld.greet()
}

fn main() {
    greet();
}
