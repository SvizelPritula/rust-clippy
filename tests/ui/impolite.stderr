error: function name is impolite
  --> $DIR/impolite.rs:4:8
   |
LL |     fn greet(&self);
   |        ^^^^^
   |
   = note: `-D clippy::impolite` implied by `-D warnings`
help: consider using a polite prefix
   |
LL |     fn please_greet(&self);
   |        ~~~~~~~~~~~~
help: consider using a polite suffix
   |
LL |     fn greet_please(&self);
   |        ~~~~~~~~~~~~

error: function name is impolite
  --> $DIR/impolite.rs:15:4
   |
LL | fn greet() {
   |    ^^^^^
   |
help: consider using a polite prefix
   |
LL | fn please_greet() {
   |    ~~~~~~~~~~~~
help: consider using a polite suffix
   |
LL | fn greet_please() {
   |    ~~~~~~~~~~~~

error: aborting due to 2 previous errors

